packages/runtime-core/src/apiLifecycle.ts

```diff
@@ -48,68 +55,37 @@ function injectHook(
   }
 }
 
-export function onBeforeMount(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.BEFORE_MOUNT, hook, target)
-}
-
-export function onMounted(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.MOUNTED, hook, target)
-}
-
-export function onBeforeUpdate(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.BEFORE_UPDATE, hook, target)
-}
-
-export function onUpdated(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.UPDATED, hook, target)
-}
-
-export function onBeforeUnmount(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.BEFORE_UNMOUNT, hook, target)
-}
-
-export function onUnmounted(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.UNMOUNTED, hook, target)
-}
-
-export function onRenderTriggered(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.RENDER_TRIGGERED, hook, target)
-}
+const createLifecycleInjector = <T extends Function = Function>(
+  lifecycle: LifecycleHooks
+): Function =>
+  function(
+    hook: T,
+    target: ComponentInternalInstance | null = currentInstance
+  ) {
+    injectHook(lifecycle, hook, target)
+  }
 
-export function onRenderTracked(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.RENDER_TRACKED, hook, target)
-}
+export const onBeforeMount = createLifecycleInjector(
+  LifecycleHooks.BEFORE_MOUNT
+)
+export const onMounted = createLifecycleInjector(LifecycleHooks.MOUNTED)
+export const onBeforeUpdate = createLifecycleInjector(
+  LifecycleHooks.BEFORE_UPDATE
+)
+export const onUpdated = createLifecycleInjector(LifecycleHooks.UPDATED)
+export const onBeforeUnmount = createLifecycleInjector(
+  LifecycleHooks.BEFORE_UNMOUNT
+)
+export const onUnmounted = createLifecycleInjector(LifecycleHooks.UNMOUNTED)
+export const onRenderTriggered = createLifecycleInjector<DebuggerEventHook>(
+  LifecycleHooks.RENDER_TRIGGERED
+)
+export const onRenderTracked = createLifecycleInjector<DebuggerEventHook>(
+  LifecycleHooks.RENDER_TRACKED
+)
 
 export function onErrorCaptured(
-  hook: (
-    err: Error,
-    instance: ComponentPublicInstance | null,
-    info: string
-  ) => boolean | void,
+  hook: ErrorCapturedHook,
   target: ComponentInternalInstance | null = currentInstance
 ) {
   injectHook(LifecycleHooks.ERROR_CAPTURED, hook, target)
```

Comment:
It's wired, I can't replace this with 

```typescript
export const onErrorCaptured = createLifecycleInjector<ErrorCapturedHook>(LifecycleHooks.ERROR_CAPTURED)
```

but local test successed

@yyx990803